### ISSUE TITLE
Timer Will not count up

### ISSUE THREAD

**Issue by Steve-GB** [AndrewMascolo/CountUpDownTimer#10]
Hello Andrew,
I'm having some issues with my code, I wonder if you could take a look please.
http://pastebin.com/8PF14JvL

This project drives an LED display with a RTC or a counting up stop watch.
The display shows 00 00 01 and doesnt show anything more, I've addedd the serial debug in to watch it and that two just shows 0:0:1 and nothing more.

It seems the counter isnt running for some readon. I have even tried the examples and they stop after showing one second. do you have any ideas ?
thanks,

**Comment by AndrewMascolo** [comment]
When did you download the library?

**Comment by AndrewMascolo** [comment]
Try it now. Redownload the library first.

**Comment by Steve-GB** [comment]
Hello.
Yes, thats done it. What was the fix ?
Thanks.

**Comment by AndrewMascolo** [comment]
Just some minor reordering in the code.

**Comment by Steve-GB** [comment]
HI, I've noticed another anomaly now.
When T.StartTimer(); is first set and the time begins to count, if a T.ResetTimer(); is then performed the timer goes back to zero and starts again as expected. if then a T.PauseTimer() is then set, instead of pausing it returns to zero again and starts counting. if a second T.PauseTimer() is set then it ignores it and if a third is set the timer pauses. if then T.ResumeTimer() is set the timer doesnt resume. A reset will work though returning to zero as expected. then if another resume is set then it counts 1 and stops. and second resume and the timer starts to count again. 

Sorry if this sounds a bit confusing, I'm using the same code as posted above.

If I reset the audrino and do not use the reset while the timer is running then all seems to be okay. It seems it doesnt like being reset whilst its running. it has to be paused then reset and then resumed.

**Comment by AndrewMascolo** [comment]
Yes because the library uses flags to start, stop and pause. If you reset midway then the flag will not be set to do the action you require without doing the process over. If you reset you must start again, if you pause you must resume.

**Comment by Steve-GB** [comment]
HI,

Yes, That’s working thanks.

What was the fix ?

**Comment by Steve-GB** [comment]
That makes sense. I'm keeping track of the state I want the timer in so if it's running and the reset is pressed, I'll issue the stop first, then reset and then start again.

Unless you want to put that in the library ?

Sent from my HTC

----- Reply message -----

**Comment by AndrewMascolo** [comment]
I will make a note of it and if more people have that issue, I will add it.

**Comment by Steve-GB** [comment]
I’ve not tried yet. But will your counter support two instances?

I will ultimately want two stop watches on the project.

**Comment by AndrewMascolo** [comment]
It should yes, there is nothing there that uses static variables.
